tensornetwork/block_sparse/utils.py

```diff
@@ -478,6 +491,12 @@ def _find_transposed_diagonal_sparse_blocks(
     block_dims (np.ndarray): 2-by-m array of matrix dimensions of each block.
   """
   flows = np.asarray(flows)
+  cacher = get_cacher()
+  if cacher.do_caching:
+    hash_val = _compute_hash(charges, flows, tr_partition, order)
+    if hash_val in cacher.cache:
+      return cacher.cache[hash_val]
```

Comment:
How will you deal with hash collisions?

Reply:
TBH I'm not sure

Reply:
could store the original value to be hashed, and compare any new object to be hashed to the stored object

Reply:
Well, having two of these collide would be a hugely annoying bug for the end user, so this'll need to be solved.

Reply:
actually, I can just do without hashing

Reply:
fixed